finetrainers/trainer.py

```diff
@@ -692,7 +925,16 @@ def _init_directories_and_repositories(self) -> None:
                 repo_id = self.args.hub_model_id or Path(self.args.output_dir).name
                 self.state.repo_id = create_repo(token=self.args.hub_token, name=repo_id).repo_id
 
-    def _model_config_call(self, fn, kwargs):
-        accepted_kwargs = inspect.signature(fn).parameters.keys()
-        kwargs = {k: v for k, v in kwargs.items() if k in accepted_kwargs}
-        return fn(**kwargs)
+    def _move_components_to_device(self):
+        if self.text_encoder is not None:
+            self.text_encoder = self.text_encoder.to(self.state.accelerator.device)
+        if self.text_encoder_2 is not None:
+            self.text_encoder_2 = self.text_encoder_2.to(self.state.accelerator.device)
+        if self.text_encoder_3 is not None:
+            self.text_encoder_3 = self.text_encoder_3.to(self.state.accelerator.device)
+        if self.transformer is not None:
+            self.transformer = self.transformer.to(self.state.accelerator.device)
+        if self.unet is not None:
+            self.unet = self.unet.to(self.state.accelerator.device)
```

Comment:
No biggie but we currently don't have any decent model that is based on unets, no? So, perhaps we can remove it altogether and support as in when needed?

Reply:
We don't support here, but I've been exploring an ImageDataset and SD1.5 on the side in a separate branch. This is because:
- HunyuanVideo can learn styles solely from images without any video training
- SD1.5 is extremely fast to test things with

I think it's okay to have here for testing purposes